jkube-kit/remote-dev/src/test/java/org/eclipse/jkube/kit/remotedev/RemoteDevelopmentServiceTest.java

```diff
@@ -90,23 +92,36 @@ void startInitiatesChildProcesses() {
       .debug("Starting port forwarder...");
   }
 
-  @Test
+  @ParameterizedTest
+  @CsvSource({
+    "remote-host, 1234, true",   // Local port is available
+    "remote-host, 1234, false"   // Local port is not available
+  })
   @DisplayName("start initiates if LocalPort for remote service are available")
-  void startInitsIfLocalPortAvailable() {
+  void startInitsIfLocalPortAvailable(String hostname, int port, boolean isLocalPortAvailable) {
     RemoteService remoteService = RemoteService.builder()
-        .hostname("remote-host").localPort(IoUtil.getFreeRandomPort()).port(1234).build();
+        .hostname(hostname)
+        .localPort(isLocalPortAvailable ? IoUtil.getFreeRandomPort() : 1234)
+        .port(port)
+        .build();
     remoteDevelopmentService = new RemoteDevelopmentService(
       logger, kubernetesClient, RemoteDevelopmentConfig.builder().remoteService(remoteService).build());
 
-    // When
     remoteDevelopmentService.start();
 
-    // Then
-    verify(logger, timeout(1000L).times(1))
-      .debug("Local port '%s' for remote service '%s:%s' is available",
-        remoteService.getLocalPort(),remoteService.getHostname(),remoteService.getPort());
-    verify(logger,  timeout(1000L).times(1))
-      .debug("Creating or replacing Kubernetes services for exposed ports from local environment");
+    if (isLocalPortAvailable) {
```

Comment:
If you have to use conditionals in test then it's probably not a good candidate for Parameterized tests